src/ui/components/DataTable/CustomizeColumns.tsx

```diff
@@ -1,134 +1,151 @@
+	};
+
+	const save = async () => {
+		const enabledColumns: string[] = columns
+				.filter(c => !c.hidden)
+				.map(c => c.key),
+			currentColumns: string[] = config.columns.map(c => c.key);
+
+		// Find columns we need to remove, and columns we need to add
+		const removeColumns: string[] = difference(currentColumns, enabledColumns),
+			addColumns: string[] = difference(enabledColumns, currentColumns);
+
+		// Apply removals and adding to currentColumns while trying to preserve the order of currentColumns
+		const nextColumns: string[] = currentColumns.filter(
+			c => !removeColumns.includes(c),
+		);
+		nextColumns.push(...addColumns);
+
+		await toWorker("main", "updateColumns", {
+			columns: nextColumns,
+			key: config.tableName,
+		});
+		onHide();
+		onSave();
+	};
+
+	const colsGrouped: { [key: string]: ColConfig[] } = groupBy(
+		columns,
+		c => c.cat,
+	);
 
 	return (
-		<Modal animation={false} centered show={show} onHide={onHide}>
-			<Modal.Header closeButton>Customize Columns</Modal.Header>
+		<Modal
+			animation={false}
+			size={"lg"}
+			show={show}
+			onEscapeKeyDown={exit}
+			centered
+		>
+			<Modal.Header onHide={exit} closeButton>
+				Customize Columns
+			</Modal.Header>
 			<Modal.Body>
-				<p>
-					Click and drag to reorder columns, or use the checkboxes to show/hide
-					columns.
-				</p>
-				{hasSuperCols ? (
-					<p className="text-danger">
-						This table has two header rows. That means you can enable/disable
-						columns, but not reorder them.
-					</p>
-				) : null}
-				<Container
-					helperClass="sort-inside-modal"
-					isDragged={isDragged}
-					onSortStart={() => {
-						setIsDragged(true);
-					}}
-					onSortEnd={args => {
-						setIsDragged(false);
-						if (!hasSuperCols) {
-							onSortEnd(args);
-						}
-					}}
-				>
-					{colOrder.map(({ colIndex, hidden }, i) => {
-						const col = cols[colIndex];
-						return (
-							<Item
-								key={colIndex}
-								index={i}
-								onToggleHidden={onToggleHidden(i)}
-								hidden={hidden}
-								col={col}
-							/>
-						);
-					})}
-				</Container>
+				<ul className="list-group list-group-flush">
+					{Object.entries(colsGrouped).map(([group, cols]) => (
+						<li key={group} className="list-group-item">
+							<strong>{group}</strong>
+							<div className="row">
+								{cols.map((col, i) => (
+									<div key={i} className="col-lg-4">
+										<div className="form-check d-flex">
+											<input
+												id={`show-column-${col.key}`}
+												className="form-check-input cursor-pointer"
+												type="checkbox"
+												checked={!col.hidden}
+												onChange={onChange(col.key)}
+											/>
+											<label
+												className="form-check-label cursor-pointer user-select-none flex-grow-1"
+												htmlFor={`show-column-${col.key}`}
+												title={col.desc}
+											>
+												{col.title}
+												<small className="d-lg-none ml-2">{col.desc}</small>
```

Comment:
Changed it to `ml-2`
In the grid form it starts wrapping and all sorts of mess. 
![image](https://user-images.githubusercontent.com/45333542/150995369-e494e149-2c27-4bd4-b8a0-e7e97c73b37b.png)
When making this I originally just had it as a title but then realized mobile/tablet users wouldnt be able to hover and see it so I added a small description for those screen sizes.

@dumbmatter What do you think the best way to handle this is?

Reply:
But I would agree that showing the description generally makes things easier to read like: 

- **Spd** _Speed_
- **AST** _Assists_ 

However, some descriptions would really need to be trimmed like:

- **AST%** _Percentage of teammate field goals a player assisted while on the floor_
- **oRtg** _Offensive Rating (points produced/scored per 100 possessions)_

Reply:
It should be possible to truncate it when it exceeds one line, using https://getbootstrap.com/docs/5.1/helpers/text-truncation/ - but it might be a bit fiddly to get working right. If you can't figure it out, just leave it as is and I'll worry about it later.

Also I guess ignore what I said about removing the `title` then, since it could be partially hidden.

For ones like Weight/Height that have the same text twice, is that an error in the column definition (useless repetition)? If so, it should be fixed there.

Reply:
For the width column I believe the column title is "Width" and the description is also "Width". I'll clean up any of those i find

Reply:
![image](https://user-images.githubusercontent.com/45333542/151395292-7c8d3d04-97c9-453f-92fd-8c22fe0e9308.png)
I got it to look pretty good, I feel like the columns without descriptions are a little weird thou but should be sufficient for now